### PR TITLE
validate cpus according to actual number of node cpus

### PR DESCRIPTION
**Pull request by rawdaGastan**
### Issues

- https://github.com/threefoldtech/zos/issues/2056

### CODE REVIEW

File: pkg/vm.go [hunk 1]
```diff
@@ -175,8 +176,16 @@ func (vm *VM) Validate() error {
 		return fmt.Errorf("invalid memory must not be less than 250M")
 	}
 
-	if vm.CPU == 0 || vm.CPU > 32 {
-		return fmt.Errorf("invalid cpu must be between 1 and 32")
+	n, err := cpu.Counts(true)
```
Comment (xmonader) [inline pkg/vm.go]: can't we use runtime.NumCPU? https://pkg.go.dev/runtime#NumCPU, if so, let's not introduce a new dependency like gopsutil

File: pkg/vm.go [hunk 2]
```diff
@@ -234,7 +243,7 @@ func (n *NetMetric) Nu() float64 {
 }
 
 // MachineMetric is a container for metrics from multiple networks
-// currently only groped as private (wiregaurd + yggdrasil), and public (public Ips)
+// currently only groped as private (wireguard + yggdrasil), and public (public Ips)
```
Comment (xmonader) [inline pkg/vm.go]: groped -> grouped

File: pkg/vm.go (hunk 1 above)
Comment (xmonader) [reply pkg/vm.go]: n -> numCpus or ncpus or cpus

Comment (muhamadazmy) [reply pkg/vm.go]: > can't we use runtime.NumCPU? https://pkg.go.dev/runtime#NumCPU, if so, let's not introduce a new dependency like gopsutil

this is not the same, according to the docs, the runtime.NumCPU is logical number of cpus that can be used by the `current` process. this is a totally Go runtime thing, not related to the total number of CPUs available on the system

Comment (muhamadazmy) [reply pkg/vm.go]: But should definitely rename the variable from `n` to something else

**Review by muhamadazmy — CHANGES_REQUESTED** [review]

**Review by muhamadazmy — APPROVED** [review]

**Pull request merged**